File: python_solutions/1249.py
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        open_ind = []
        delete_idx = set()
        for idx,c in enumerate(s):
            if c == '(':
                open_ind.append(idx)
            elif c == ')':
                if len(open_ind) > 0:
                    open_ind.pop()
                else:
                    delete_idx.add(idx)
        while len(open_ind) > 0:
            delete_idx.add(open_ind.pop())
        new_str = ""
        for idx,c in enumerate(s):
            if idx not in delete_idx:
                new_str += c
        return new_str
        

File: rust_solutions/1249.rs
impl Solution {
    pub fn min_remove_to_make_valid(s: String) -> String {
        use std::collections::HashSet ; 
        let mut open_ind = Vec::new();
        let mut delete_idx = HashSet::new();
        
        for (idx,c) in s.chars().enumerate() {
            match c {
                '(' => open_ind.push(idx),
                ')' => {
                    match open_ind.len() {
                        0 => {delete_idx.insert(idx);},
                        _ => {open_ind.pop();}
                    }
                },
                _ => ()
            }
        }
        for idx in open_ind.iter() {
            delete_idx.insert(*idx);
        }
        
        let mut new_str = String::new();
        for (idx , c) in s.chars().enumerate() {
            if !delete_idx.contains(&idx) {
                new_str.push(c);
            }
        }
        new_str
    }
}
